docs: e2e test restructuring ADR (#2068)
kyma-project/telemetry-manager: docs/contributor/arch/020-redesign-e2e-log-test-structure.md

  - Custom filters
  - Custom outputs
  - Custom parsers
- LogAgent:
  - TraceID/SpanID extraction
- LogGateway:
  - Manual gateway scaling

#### Miscellaneous
- Version conversion tests (e.g., v1alpha1 → v1beta1)

Here’s a suggested file structure:

```bash
/test
├── e2e
│   ├── shared/ # Shared tests (all 3 components, FluentBit/LogAgent, LogGateway/LogAgent) implemented as table-driven
│   │   ├── namespace_selector_test.go # all 3 components
│   │   ├── container_selector_test.go # FluentBit/LogAgent (LogGateway does not support this)
│   │   ├── k8s_attr_enrichment.go # LogGateway/LogAgent (FluentBit does not support this)
│   │   ├── service_name_enrichment.go # LogGateway/LogAgent (FluentBit does not support this)
│   │   ├── mtls_test.go # all 3 components
│   │   ├── self_monitoring_test.go # all 3 components
│   │   └── validation_test.go # all 3 components
│   │
│   ├── fluentbit/
│   │   ├── dedotting_test.go
│   │   ├── custom_output_test.go
│   │
│   ├── logagent/
│   │   ├── trace_attribute_parser_test.go
│   │
│   ├── loggateway/
│   │   ├── otlp_push_test.go
│   │   └── manual_scaling_test.go
│   │
│   └── conversion/
│   │   └── version_conversion_test.go
```

> Shared tests will be implemented as table-driven tests, which will allow us to easily add new test cases and configurations without duplicating code.

Reviewer: ```suggestion
Shared tests will be implemented as table-driven tests, so we can easily add new test cases and configurations without duplicating code.
```
